rwengine/src/engine/SaveGame.cpp

```diff
@@ -1189,6 +1189,24 @@ bool SaveGame::loadGame(GameState& state, const std::string& file)
 		vehicle->setSecondaryColour(car.colorBG);
 	}
 
+	for (int g = 0; g < carGenerators.size(); ++g) {
+		auto& gen = carGenerators[g];
+		state.vehicleGenerators.push_back({
```

Comment:
Actually.. can't you just add a function to return a proper VG from the SaveGame VG type? (Up to you)